palantir-java-format/src/test/resources/com/palantir/javaformat/java/testdata/palantir-10.output

```diff
@@ -0,0 +1,10 @@
+class Palantir10 {
+    @SqlUpdate("CREATE TABLE things "
+            + "(id VARCHAR(" + MAX_LENGTH + ") NOT NULL, "
+            + "title VARCHAR(" + MAX_LENGTH + ")  NOT NULL, "
+            + "description VARCHAR(" + MAX_DESCRIPTION_LENGTH + ") NOT NULL, "
+            + "standard BOOLEAN NOT NULL, "
+            + "documentJson VARCHAR(" + MAX_DOCUMENT_LENGTH + ") NOT NULL, "
+            + "PRIMARY KEY (id))")
```

Comment:
I wonder if we're going to lose some otherwise nice reformatting by optimizing for this case where there are 14 `+` symbols... like what about short snippets with just a little concatenation?

```java
doSomething("foo" +
 bar + "baz")
```

Maybe we should count the +s?

Reply:
The formatter seems to behave well in small cases as well. It allowed
```
class Palantir10 {
    @SqlUpdate("CREATE TABLE things " + MAX_LENGTH + ")")
    void createTable();
}
```

Reply:
ok, how about this - one newline => don't bother, more => keep

Reply:
added tests to showcase @iamdanfox's  concern is now handled well

Reply:
@ferozco the small case you pasted worked before as well, since the annotation fits onto one line there is no need to ever split it.
This is only tricky when the whole annotation can't fit on a single line.